src/modal.rs

```diff
@@ -2,11 +2,12 @@ use crate::widget::Element;
 use data::{config, Server};
 
 pub mod connect_to_server;
-pub mod reload_configuration_error;
+pub mod error;
 
 #[derive(Debug)]
 pub enum Modal {
     ReloadConfigurationError(config::Error),
+    IrcError(anyhow::Error),
```

Comment:
I'm not sure I'm sold on putting errors in a modal. It seems very disruptive. We have a log buffer now, we should just make sure everything is logged and they can be viewed there. 

Are there certain types of errors that need to escalate to a modal? We can probably think of a different UI mechanism that indicates errors have occurred in the sidebar and clicking it opens the log buffer

Reply:
i was thinking that an error that opens up in a corner window would be the best, not one that takes up full screen like the current modal

Reply:
I think we should use the logger buffer for this.

Reply:
So like a toast notification? I'll defer to @casperstorm on that. Anything that bleeds over the panes will be quite disruptive / noisy imo. 

I'm thinking we show an error icon in the sidebar, clicking it clears the icon and opens log buffer. New errors make icon reappear if log buffer is closed. And icon visibility is configurable.

Reply:
I was thinking a toast in the sidebar that you can click away

but a link that goes to the log buffer would also be good IMO.

For now I'll log an error and we can work on the UI element later

Reply:
Sounds good 👍